app/controlplane/pkg/ca/fileca/fileca.go

```diff
@@ -0,0 +1,47 @@
+//
+// Copyright 2024 The Chainloop Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fileca
+
+import (
+	"context"
+	"crypto/x509"
+	"fmt"
+
+	"github.com/chainloop-dev/chainloop/app/controlplane/pkg/ca"
+	fulcioca "github.com/sigstore/fulcio/pkg/ca"
+	fulciofileca "github.com/sigstore/fulcio/pkg/ca/fileca"
+	"github.com/sigstore/fulcio/pkg/identity"
+)
+
+type FileCA struct {
+	ca fulcioca.CertificateAuthority
+}
+
+var _ ca.CertificateAuthority = (*FileCA)(nil)
+
+func NewFileCA(certPath, keyPath, keyPass string, watch bool) (*FileCA, error) {
```

Comment:
since you are doing some refactoring, you might want to call this just `New` as you do in the other provider.